text/0000-third-party-web-components-support.md

```diff
@@ -0,0 +1,155 @@
+with this directive. If it happens that the component was not registered, the custom element will
+simply render as an HTMLUnknownElement, which is the default browser behavior.
+
+For example, given the following template:
+```html
+<template>
+  <lwc-foo bar={bar}></lwc-foo>
+  <native-foo lwc:external bar={bar}></native-foo>
+</template>
+```
+
+The template compiler would output the following:
+```js
+import _lwcFoo from "lwc/foo";
+import { registerTemplate } from "lwc";
+function tmpl($api, $cmp, $slotset, $ctx) {
+  const { c: api_custom_element, h: api_element } = $api;
+  return [
+    api_custom_element("lwc-foo", _lwcFoo, {
+      props: {
+        foo: $cmp.bar,
+      },
+      key: 0,
+    }),
+    api_element("native-foo", {
+      attrs: {
+        foo: $cmp.bar,
+      },
+      key: 1,
+    }),
+  ];
+}
+export default registerTemplate(tmpl);
+tmpl.stylesheets = [];
+```
+
+### Passing data
+
+By convention, custom elements are assumed to be backed by LWC components and values are set through
+their properties. Given that a native web component might expect data passed as an attribute or as a
+property, we would need to support both mechanisms. There are multiple approaches we could take
+here, and the following options are how some popular frameworks do it today. We could default to
+either attributes or properties and introduce a directive that would act as a signal to the engine
+to use one or the other, we could implement a runtime heuristic to choose one over the other, or we
+could default to attributes and set properties when setting a data structure.
+
+#### Proposed solution
+
+When passing data to a native web component, we can take a heuristical approach where if the
+property exists, the property is set. Otherwise the attribute is set. While it may be advantageous
+to require an explicit signal in the form of a directive (both in terms of static analysis and
+performance), the proposed solution strikes a good balance between developer experience and API
+semantics. This heuristical approach will require a check on the element's prototype to see whether
+a property is defined. This will incur a runtime performance cost, which can be amortized over the
+lifetime of the component by caching the result of the check.
+
+### Events
+
+Today, LWC's declarative event bindings only support lowercase events. Events using anything other
+than lowercase can be listened for using the imperative `addEventListener()` API. There are several
+historical discussions (e.g., [#1811][1811] and [#1904][1904]) about relaxing this constraint;
+however, the current behavior would not need to change in order to support native web components. It
+should be noted that adding support for events has always been about improved interoperability, as
+the browser's imperative API has no restriction regarding the event type.
+
+EventTarget polyfilled methods and properties such as `add/removeEventListener` and `target` go

```

Comment:
I didn't quite get what you're saying here.

Reply:
I'm basically saying that I investigated a little to see if the synthetic shadow polyfill for events might need to be adjusted and found that no work was required there.